controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1385,10 +1480,19 @@ func (r *BareMetalHostReconciler) manageHostPower(prov provisioner.Provisioner,
 			return actionContinue{}
 		}
 	}
+	hup, err := r.acquireHostUpdatePolicy(info)
+	if err != nil {
+		info.log.Info("failed setting owner reference on hostupdatepolicy")
+		return actionError{errors.Wrap(err, "failed setting owner reference on hostUpdatePolicy")}
+	}
 
-	provState := info.host.Status.Provisioning.State
-	// Normal reboots only work in provisioned states, changing online is also possible for available hosts.
-	isProvisioned := provState == metal3api.StateProvisioned || provState == metal3api.StateExternallyProvisioned
+	servicingAllowed := isProvisioned && !info.host.Status.PoweredOn && desiredPowerOnState
```

Comment:
nit: there is already an `if !info.host.Status.PoweredOn` block above. We could move all of this new code inside that.

Reply:
Something like https://paste.opendev.org/show/bAWRmUqzLC5hrAbS8Uec/ ?

Reply:
The annotation has to be deleted before servicing, and you no longer need acquireHostUpdatePolicy if you do it on registration.

Reply:
Okay, I guess it won't hurt to call it here as well.